apispec_webframeworks/flask.py

```diff
@@ -114,3 +153,38 @@ def path_helper(self, operations, view, **kwargs):
                     method = getattr(view.view_class, method_name)
                     operations[method_name] = yaml_utils.load_yaml_from_docstring(method.__doc__)
         return self.flaskpath2openapi(rule.rule)
+
+
+class DocumentedBlueprint(Blueprint):
+    """Flask Blueprint which documents every view function defined in it."""
+
+    def __init__(self, name, import_name, spec):
+        """
+        Initialize blueprint. Must be provided an APISpec object.
+        :param APISpec spec: APISpec object which will be attached to the blueprint.
+        """
+        super(DocumentedBlueprint, self).__init__(name, import_name)
+        self.documented_view_functions = defaultdict(list)
+        self.spec = spec
+
+    def route(self, rule, documented=True, **options):
+        """If documented is set to True, the route will be added to the spec.
+        :param bool documented: Whether you want this route to be added to the spec or not.
+        """
+
+        return super(DocumentedBlueprint, self).route(rule, documented=documented, **options)
+
+    def add_url_rule(self, rule, endpoint=None, view_func=None, documented=True, **options):
+        """If documented is set to True, the route will be added to the spec.
+        :param bool documented: Whether you want this route to be added to the spec or not.
+        """
+        super(DocumentedBlueprint, self).add_url_rule(rule, endpoint=endpoint, view_func=view_func, **options)
+        if documented:
+            self.documented_view_functions[rule].append(view_func)
+
+    def register(self, app, options, first_registration=False):
```

Comment:
For my usercase I prefer to pass the spec in the register (or add_url_rule method) instead of the constructor.

Reply:
That could be a solution, yeah!

`app.register_blueprint(doc_blueprint, spec=spec)`

I'd still add the option of passing it through the `__init__` as a kwarg and add an `attach(spec)` method, to add variety of uses.

What do you think @tinproject ?